kubernetes/resource_kubernetes_persistent_volume_claim.go

```diff
@@ -178,32 +176,22 @@ func resourceKubernetesPersistentVolumeClaimCreate(d *schema.ResourceData, meta
 					return out, "", err
 				}
 
-				events, err := conn.CoreV1().Events(metadata.Namespace).List(meta_v1.ListOptions{
-					FieldSelector: fields.Set(map[string]string{
-						"involvedObject.name":      metadata.Name,
-						"involvedObject.namespace": metadata.Namespace,
-						"involvedObject.kind":      "PersistentVolumeClaim",
-					}).String(),
-				})
-				if err != nil {
-					return out, "", err
-				}
-				if len(events.Items) > 0 {
-					lastEvent = events.Items[0]
-				}
-
 				statusPhase := fmt.Sprintf("%v", out.Status.Phase)
 				log.Printf("[DEBUG] Persistent volume claim %s status received: %#v", out.Name, statusPhase)
 				return out, statusPhase, nil
 			},
 		}
 		_, err = stateConf.WaitForState()
 		if err != nil {
-			reason := ""
-			if lastEvent.Reason != "" {
-				reason = fmt.Sprintf(". Reason: %s: %s", lastEvent.Reason, lastEvent.Message)
+			warns := ""
+			lastWarnings, wErr := getLastWarningsForObject(conn, out.ObjectMeta, "PersistentVolumeClaim", 3)
+			if wErr != nil {
+				return wErr
+			}
+			if len(lastWarnings) > 0 {
+				warns = fmt.Sprintf(". Last warnings: %s", stringifyEvents(lastWarnings))
```

Comment:
I'd argue this would read cleaner if only returning the combined err + warns object within this if statement?